sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/ClientRetryPolicy.java

```diff
@@ -161,6 +165,28 @@ public Mono<ShouldRetryResult> shouldRetry(Exception e) {
                 clientException);
         }
 
+        if (clientException != null && Exceptions.isStatusCode(clientException, HttpConstants.StatusCodes.REQUEST_TIMEOUT)) {
+            logger.info(
+                "Request timeout - IsReadRequest {}, IsWebExceptionRetriable {}, NonIdempotentWriteRetriesEnabled {}",
+                this.isReadRequest,
+                false,
+                this.request.getNonIdempotentWriteRetriesEnabled(),
+                e);
+
+            return this.shouldRetryOnRequestTimeout(
+                this.isReadRequest,
+                this.request.getNonIdempotentWriteRetriesEnabled()
+            );
+        }
+
+        if (clientException != null &&
+            Exceptions.isStatusCode(clientException, HttpConstants.StatusCodes.INTERNAL_SERVER_ERROR) &&
+            Exceptions.isSubStatusCode(clientException, HttpConstants.SubStatusCodes.UNKNOWN)) {
```

Comment:
should we treat the same for all 500/*?

Reply:
My thought process was to scope it to 500s bubbled up by the service. I could make a case for inclusion in circuit breaking flow for deserializing service-side response which the SDK wraps as a 500 but not sure about invalid session token parsing which can also bubble up 500s but from the client.

Reply:
For 500s - in the JVM sync we decided to not special case by sub-status code - since the weird behavior at this point can mostly be attributed to server-side issues.